core/ide/src/main/java/org/alice/ide/ProjectApplication.java

```diff
@@ -352,40 +365,210 @@ public UserActivity getOpenActivity() {
   }
 
   public final void loadProject(UserActivity activity, UriProjectLoader uriProjectLoader) {
+    loadProject(activity, uriProjectLoader, false, new HashSet<>());
+  }
+
+  private final void loadProject(UserActivity activity, UriProjectLoader uriProjectLoader, boolean isLoadingBackups,
+                                Set<String> unloadableFiles) {
     showWaitCursor();
     this.uriProjectLoader = uriProjectLoader;
     if (uriProjectLoader != null) {
-      uriProjectLoader.deliverContentOnEventDispatchThread(proj -> projectLoaded(activity, proj));
+      uriProjectLoader.deliverContentOnEventDispatchThread(proj -> projectLoaded(activity, proj, isLoadingBackups,
+              unloadableFiles));
     }
   }
 
-  private void projectLoaded(UserActivity activity, Project project) {
+  private void projectLoaded(UserActivity activity, Project project, boolean isLoadingBackups,
+                             Set<String> unloadableFiles) {
+    File saved = UriUtilities.getFile(getUri());
+
+    boolean isBackup = false;
+
+    if (!projectFileUtilities.isNewProject()) {
+      File parentDir = saved.getParentFile();
+
+      if (parentDir != null) {
+        isBackup = BACKUP_EXTENSION.equals(getExtension(parentDir.getName()));
+      }
+    }
+
+    //File backupDir = projectFileUtilities.backupDirectory(saved, isBackup).toFile();
+    //boolean makeVrReady = uriProjectLoader.shouldMakeVrReady();
+
     if (project == null) {
-      uriProjectLoader = null;
-      activity.cancel();
+      handleProjectLoadError(saved, activity, isBackup, isLoadingBackups, unloadableFiles);
     } else {
-      try {
-        updateInterface(project);
-      } catch (RuntimeException re) {
-        var message = new StringBuilder("Errors reported in " + getUri());
-        Throwable cause = re;
-        do {
-          var causeMessage = cause.getLocalizedMessage();
-          if (causeMessage != null) {
-            message.append("\n\n  ").append(causeMessage);
-          }
-          cause = cause.getCause();
-        } while (cause != null);
-        // TODO clear project remnants from system
+      handleProjectLoadSuccess(project, saved, activity, isBackup, isLoadingBackups, unloadableFiles);
+    }
+    hideWaitCursor();
```

Comment:
Having this here means it only happens if `uriProjectLoader` is not null (line 375)
The show and hide should be more tightly coupled, perhaps with the hide in a finally block to make sure it happens.